src/renderer/components/swap/Swap.tsx

```diff
@@ -390,6 +389,156 @@ export const Swap = ({
+  )
+
+  const sourceChainErrorLabel: JSX.Element = useMemo(() => {
+    if (!sourceChainBalanceError) {
+      return <></>
+    }
+    const sourceChainAsset = FP.pipe(
+      sourceAsset,
+      O.map(({ chain }) => getChainAsset(chain)),
+      O.map((asset) => asset.symbol),
+      O.getOrElse(() => '')
+    )
+
+    return (
+      <Styled.BalanceErrorLabel>
+        {intl.formatMessage({ id: 'swap.errors.amount.balanceShouldCoverChainFee' }, { asset: sourceChainAsset })}{' '}
+      </Styled.BalanceErrorLabel>
+    )
+  }, [sourceChainBalanceError, sourceAsset, intl])
+
+  const targetChainFeeAmountInTargetAsset: BaseAmount = useMemo(() => {
+    const chainFee = FP.pipe(
+      targetChainFee,
+      RD.getOrElse(() => ZERO_BASE_AMOUNT)
+    )
+
+    return FP.pipe(
+      targetAsset,
+      O.map((asset) => {
+        const chainAsset = getChainAsset(asset.chain)
+        return getValueOfAsset1InAsset2(chainFee, poolData[assetToString(chainAsset)], poolData[assetToString(asset)])
+      }),
+      O.getOrElse(() => ZERO_BASE_AMOUNT)
+    )
+  }, [targetChainFee, targetAsset, poolData])
+
+  const targetChainFeeError = useMemo((): boolean => {
+    if (changeAmount.isZero()) {
+      return false
+    }
+    const targetFee = FP.pipe(targetChainFeeAmountInTargetAsset, baseToAsset, (assetAmount) => assetAmount.amount())
+    return swapData.swapResult.minus(targetFee).isNegative()
+  }, [targetChainFeeAmountInTargetAsset, swapData, changeAmount])
+
+  const targetChainFeeErrorLabel = useMemo(() => {
+    if (!targetChainFeeError) {
+      return null
+    }
+
+    const feeAssetAmount = baseToAsset(targetChainFeeAmountInTargetAsset)
+
+    const feeLabel = FP.pipe(
+      targetAsset,
+      O.map((asset) => formatAssetAmountCurrency({ amount: feeAssetAmount, asset, trimZeros: true })),
+      O.getOrElse(() => formatBN(feeAssetAmount.amount()))
+    )
+
+    return (
+      <Styled.ErrorLabel>
+        {intl.formatMessage({ id: 'swap.errors.amount.outputShouldCoverChainFee' }, { fee: feeLabel })}
+      </Styled.ErrorLabel>
+    )
+  }, [targetChainFeeError, targetChainFeeAmountInTargetAsset, intl, targetAsset])
+
+  const fees: RD.RemoteData<Error, Fee[]> = useMemo(
+    () =>
+      FP.pipe(
+        sequenceTRD(
+          sourceChainFee,
+          RD.success(targetChainFeeAmountInTargetAsset),
+          RD.fromOption(sourceAsset, () => Error('No source asset')),
+          RD.fromOption(targetAsset, () => Error('No target asset'))
+        ),
+        RD.map(([sourceFee, targetFee, sourceAsset, targetAsset]) => [
+          { asset: getChainAsset(sourceAsset.chain), amount: sourceFee },
+          { asset: targetAsset, amount: targetFee }
+        ])
+      ),
+    [sourceChainFee, targetChainFeeAmountInTargetAsset, sourceAsset, targetAsset]
+  )
+
+  const isSwapDisabled: boolean = useMemo(
+    () => changeAmount.eq(0) || FP.pipe(walletBalances, O.isNone) || sourceChainBalanceError || targetChainFeeError,
```

Comment:
Don't disable swap in case of `sourceChainBalanceError` or `targetChainFeeError`